ast/src/analyzed/mod.rs

```diff
@@ -186,6 +194,28 @@ impl<T> Analyzed<T> {
             .push(StatementIdentifier::Identity(id as usize));
         id
     }
+
+    /// Remove some identities by their index (not their ID).
+    /// Does not re-allocate IDs.
+    pub fn remove_identities(&mut self, to_remove: &BTreeSet<usize>) {
+        let mut shift = 0;
+        self.source_order.retain_mut(|s| {
+            if let StatementIdentifier::Identity(index) = s {
+                if to_remove.contains(index) {
+                    shift += 1;
+                    return false;
+                }
+                *index -= shift;
+            }
+            true
+        });
+        let mut index = 0;
+        self.identities.retain(|_| {
+            let retain = !to_remove.contains(&index);
+            index += 1;
+            retain
+        })
+    }
```

Comment:
Wasn't this (or something similar) part of another PR?

Reply:
Yes, I had another PR for add_identity and remove_column above.

Reply:
no, it's just similar